Update programsblock and programcard's theme

ivanleeswe opened the pull request:
Fix #273

ivanleeswe committed:
Update programsblock and programcard's theme

richardxia commented:
Hey @ivanleeswe, sorry if the ticket wasn't clear, but I think it'd be a good idea to make the the background color configurable via a React prop, similar to all the other similar tickets. For example, see @candywang's older PR https://github.com/ShelterTechSF/sheltertech.org/pull/278 or @davidagustin's open PR https://github.com/ShelterTechSF/sheltertech.org/pull/286.

Do you think you could add a prop to choose between a white background and a light gray background like those PRs?

richardxia committed:
Add theme prop to ProgramsBlock, and remove theme prop from ProgramCard.

Since the only remaining thing the theme prop on ProgramCard affected
was the font color, and since the site now sticks to lighter shades for
backgrounds, the prop now has no use.

richardxia commented:
I updated this PR by rebasing it on `main` and adding a `theme` prop to the `<ProgramsBlock>` component, similar to the other components we've been updating. I also ended up removing the `theme` prop from the `<ProgramCard>` component, since the only remaining thing it affected was the text color, which no longer needs to be configurable because the components always have a lighter background and therefore the text color can stay with the site's default darker color.

If anyone wants to take a quick look at this, that'd be great, but otherwise I'll probably merge it in a few days just so that we can do a push to prod by the end of the year, with all of the other style updates.

Thanks again, @ivanleeswe, for starting the work on this PR!

richardxia commented:
Going to merge this in, as to not block shipping the overall updates to the page.
